test/unittest/QosSerializer/QosSerializerTests.cpp

```diff
@@ -0,0 +1,782 @@
+    EXPECT_EQ(expected, serialized);
+
+    qos.kind =  MANUAL_BY_PARTICIPANT_LIVELINESS_QOS;
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    expected[field.c_str()]["kind"] = "MANUAL_BY_PARTICIPANT_LIVELINESS_QOS";
+    EXPECT_EQ(expected, serialized);
+
+    qos.kind =  MANUAL_BY_TOPIC_LIVELINESS_QOS;
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    expected[field.c_str()]["kind"] = "MANUAL_BY_TOPIC_LIVELINESS_QOS";
+    EXPECT_EQ(expected, serialized);
+}
+
+TEST(qos_serializer_tests, reliability_qos_policy)
+{
+    ReliabilityQosPolicy qos;
+    Qos serialized;
+    Qos expected;
+    std::string field("field");
+
+    qos.kind =  BEST_EFFORT_RELIABILITY_QOS;
+    qos.max_blocking_time = eprosima::fastrtps::Duration_t (10, 10);
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    Qos serialized_blocking_time;
+
+    eprosima::statistics_backend::subscriber::serialize(qos.max_blocking_time, "max_blocking_time", serialized_blocking_time);
+    expected[field.c_str()]["kind"] = "BEST_EFFORT_RELIABILITY_QOS";
+    expected[field.c_str()]["max_blocking_time"] = serialized_blocking_time["max_blocking_time"];
+    EXPECT_EQ(expected, serialized);
+
+    qos.kind =  RELIABLE_RELIABILITY_QOS;
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    expected[field.c_str()]["kind"] = "RELIABLE_RELIABILITY_QOS";
+    EXPECT_EQ(expected, serialized);
+}
+
+TEST(qos_serializer_tests, ownership_qos_policy)
+{
+    OwnershipQosPolicy qos;
+    Qos serialized;
+    Qos expected;
+    std::string field("field");
+
+    qos.kind =  SHARED_OWNERSHIP_QOS;
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    expected[field.c_str()]["kind"] = "SHARED_OWNERSHIP_QOS";
+    EXPECT_EQ(expected, serialized);
+
+    qos.kind =  EXCLUSIVE_OWNERSHIP_QOS;
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    expected[field.c_str()]["kind"] = "EXCLUSIVE_OWNERSHIP_QOS";
+    EXPECT_EQ(expected, serialized);
+}
+
+TEST(qos_serializer_tests, destination_order_qos_policy)
+{
+    DestinationOrderQosPolicy qos;
+    Qos serialized;
+    Qos expected;
+    std::string field("field");
+
+    qos.kind =  BY_RECEPTION_TIMESTAMP_DESTINATIONORDER_QOS;
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    expected[field.c_str()]["kind"] = "BY_RECEPTION_TIMESTAMP_DESTINATIONORDER_QOS";
+    EXPECT_EQ(expected, serialized);
+
+    qos.kind =  BY_SOURCE_TIMESTAMP_DESTINATIONORDER_QOS;
+    eprosima::statistics_backend::subscriber::serialize(qos, field, serialized);
+
+    expected[field.c_str()]["kind"] = "BY_SOURCE_TIMESTAMP_DESTINATIONORDER_QOS";
+    EXPECT_EQ(expected, serialized);
+}
+
+TEST(qos_serializer_tests, topic_data_qos_policy)
+{
+    TopicDataQosPolicy qos(PID_PAD);
```

Comment:
There is no need to pass `PID_PAD` as argument. It is enough with the default constructor. This also applies to the next two tests.

```
    TopicDataQosPolicy qos;
```